bundles/edu.kit.ipd.sdq.commons.util.eclipse/src/edu/kit/ipd/sdq/commons/util/org/eclipse/core/resources/IProjectUtil.xtend

```diff
@@ -71,13 +72,18 @@ class IProjectUtil {
 	 * @throws IllegalStateException if the project already exists
 	 * @throws CoreException if configuring the project fails
 	 */
-	def static configureAsJavaProject(IProject project) {
+	def static IJavaProject configureAsJavaProject(IProject project) {
 		// copied from:
 		// https://sdqweb.ipd.kit.edu/wiki/JDT_Tutorial:_Creating_Eclipse_Java_Projects_Programmatically
 		project.open(new NullProgressMonitor())
 		val description = project.getDescription()
-		description.setNatureIds(#{JavaCore.NATURE_ID})
+		val currentNatureIds = description.natureIds
+		val newNatureIds = newArrayOfSize(currentNatureIds.length + 1)
+		System.arraycopy(currentNatureIds, 0, newNatureIds, 0, currentNatureIds.length)
+		newNatureIds.set(currentNatureIds.length, JavaCore.NATURE_ID)
```

Comment:
I am not sure what happens if the project is already configured as a Java project (i.e., if it, in particular, already has the Java nature). Will the nature be duplicated? Maybe some subsequent statement will fail (e.g., creating the source folder). I am also not sure whether the documentation for `IllegalStateExeception` is still valid, or whether this is some outdated documentation from a time at which the method also created the project. 
Maybe we could/should add some checks at the beginning of the method about whether the project exists and whether it is not a Java project yet, such that we explicitly know (and can document) under which conditions which kind of exception is throws? But maybe it already obvious and I am only missing something.

Reply:
I see your point. The method started out as a simple creator for a whole Java project and is now being used in all kind of contexts. I’ll update it.

Reply:
Please have a look at the updated version!